src/hooks/useVotedRepos.ts

```diff
@@ -0,0 +1,56 @@
+import { useCallback, useEffect, useState } from "react";
+import { User } from "@supabase/supabase-js";
+import useSupabaseAuth from "./useSupabaseAuth";
+import { fetchRecommendations } from "../lib/supabase";
+import handleVoteUpdateByRepo from "../lib/handleVoteUpdateByRepo";
+
+const useVotedRepos = () => {
+  const [votedReposIds, setVotedReposIds] = useState<number[]>([]);
+  const { signIn, user } = useSupabaseAuth();
+
+  const fetchVotedData = useCallback(async (user?: User) => {
+    try {
+      if (user) {
+        const data = await fetchRecommendations("myVotes", 1000, user, "");
+
+        return setVotedReposIds(data.map(({ id }) => id));
+      }
+    } catch (e) {
+      console.log(e);
+    }
+
+    setVotedReposIds([]);
+  }, []);
```

Comment:
Should the data.votes be set here as the watch data?

Reply:
thought about it, but then all the vote functions we are re-using modify the `votedReposIds` globally, so we only need to fetch the recommendations on load - it will make even more sense with SWR or "my votes call" and local storage cache;

That would make the votes available on refresh even without fetching the data again, and provide integrity with voted data throughout the UI, still, not hitting the SWR revalidation.

That's the plan, at least! 🍕